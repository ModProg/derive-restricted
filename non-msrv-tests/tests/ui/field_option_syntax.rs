#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
struct Test1<T>(#[derive_where] T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
struct Test2<T>(#[derive_where = "default"] T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
enum Test3<T> {
	#[derive_where]
	A(T),
}

#[derive(derive_where::DeriveWhere)]
#[derive_where(Default; T)]
enum Test4<T> {
	#[derive_where = "default"]
	A(T),
}

fn main() {}
