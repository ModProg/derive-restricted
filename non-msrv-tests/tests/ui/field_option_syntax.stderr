error: unexpected option syntax
 --> tests/ui/field_option_syntax.rs:3:19
  |
3 | struct Test1<T>(#[derive_where] T);
  |                   ^^^^^^^^^^^^

error: unexpected option syntax
 --> tests/ui/field_option_syntax.rs:7:19
  |
7 | struct Test2<T>(#[derive_where = "default"] T);
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/field_option_syntax.rs:12:4
   |
12 |     #[derive_where]
   |       ^^^^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/field_option_syntax.rs:19:4
   |
19 |     #[derive_where = "default"]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^
